spec/requests/api/v0/classified_listings_spec.rb

```diff
@@ -0,0 +1,487 @@
+    include_context "with 7 listings and 2 user"
+
+    it "returns json response and ok status" do
+      get api_classified_listings_path
+      expect(response.content_type).to eq("application/json")
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "returns listings created" do
+      get api_classified_listings_path
+      expect(json_response.size).to eq(7)
+      expect(json_response.first["type_of"]).to eq("classified_listing")
+      expect(json_response.first["slug"]).to eq(ClassifiedListing.last.slug)
+      expect(json_response.first["user"]).to include("username")
+      expect(json_response.first["user"]["username"]).not_to be_empty
+    end
+
+    it "supports pagination" do
+      get api_classified_listings_path, params: { page: 2, per_page: 2 }
+      expect(json_response.length).to eq(2)
+      get api_classified_listings_path, params: { page: 4, per_page: 2 }
+      expect(json_response.length).to eq(1)
+    end
+  end
+
+  describe "GET /api/listings/category/:category" do
+    include_context "with 7 listings and 2 user"
+
+    it "displays only listings from the cfp category" do
+      get api_classified_listings_category_path("cfp")
+      expect(response).to have_http_status(:ok)
+      expect(json_response.size).to eq(3)
+    end
+  end
+
+  describe "GET /api/listings/:id" do
+    include_context "with 7 listings and 2 user"
+    let(:listing) { ClassifiedListing.where(category: "cfp").last }
+
+    it "displays a unique listing" do
+      get api_classified_listing_path(listing.id)
+      expect(response).to have_http_status(:ok)
+      expect(json_response["type_of"]).to eq("classified_listing")
+      expect(json_response["slug"]).to eq(listing.slug)
+      expect(json_response["user"]).to include("username")
+      expect(json_response["user"]["username"]).not_to be_empty
+    end
+  end
+
+  describe "POST /api/listings" do
+    def post_classified_listing(key: api_secret.secret, **params)
+      headers = { "api-key" => key, "content-type" => "application/json" }
+      post api_classified_listings_path, params: { classified_listing: params }.to_json, headers: headers
+    end
+
+    def user_admin_organization(user)
+      org = create(:organization)
+      create(:organization_membership, user_id: user.id, organization_id: org.id, type_of_user: "admin")
+      org
+    end
+
+    describe "user cannot proceed if not properly unauthorizedœ" do
+      let(:api_secret) { create(:api_secret) }
+
+      it "fails with no api key" do
+        post api_classified_listings_path, headers: { "content-type" => "application/json" }
+        expect(response). to have_http_status(:unauthorized)
+      end
+
+      it "fails with the wrong api key" do
+        post api_classified_listings_path, headers: { "api-key" => "foobar", "content-type" => "application/json" }
+        expect(response). to have_http_status(:unauthorized)
+      end
+
+      it "fails with a failing secure compare" do
```

Comment:
I don't think this is needed, you're testing the internals of the authentication here :D